internal/config/config.go

```diff
@@ -16,30 +40,59 @@ type Args struct {
 	Length   int     `yaml:"Length" validate:"optional"`
 }
 
-type Config struct {
-	DbDSN        string          `yaml:"DbDSN" validate:"required"`
-	DbTable      string          `yaml:"DbTable" validate:"required"`
-	TotalRecords int             `yaml:"TotalRecords" validate:"required,gt=0"`
-	BatchSize    int             `yaml:"BatchSize" validate:"required,gt=0"`
-	SafeImport   bool            `yaml:"SafeImport"`
-	Columns      map[string]Args `yaml:"Columns" validate:"required,dive,keys,required,endkeys"`
+// TableDef describes one particular database table. Its data is (mostly) loaded from a YAML file.
+type TableDef struct {
+	TableName    string               `yaml:"TableName" validate:"required"`
+	TotalRecords int                  `yaml:"TotalRecords" validate:"required,gt=0"`
+	BatchSize    int                  `yaml:"BatchSize" validate:"required,gt=0"`
+	SafeImport   bool                 // TODO: should this be global?
+	Columns      map[string]ColumnDef `yaml:"Columns" validate:"required,dive,keys,required,endkeys"`
 }
 
-func LoadConfig(filename string) (*Config, error) {
-	yamlFile, err := ioutil.ReadFile(filename)
+func LoadConfig(args RunArgs) (string, []*TableDef, error) {
+	dsn := args.GetDSN()
+	tables := make([]*TableDef, 0)
+	validate := validator.New()
+
+	err := validate.Struct(args)
 	if err != nil {
-		return nil, err
+		reportValidationErrors(err)
+		return dsn, tables, err
 	}
-	c := &Config{}
-	err = yaml.Unmarshal(yamlFile, &c)
-	if err != nil {
-		return nil, err
+
+	// TODO: allow for entire folders to be passed in and implement searching for YAML files in them
```

Comment:
Re: TODO
This is already trivially solved with bash wildcard selects. If we want everything in `table_configs` we can simply
```shell
go run ./cmd/syndi/main.go -- ./table_configs/*
```
which bash will automatically expand into all files in that folder. There's probably little value in adding this functionality.

Reply:
true... i was thinking more about walking subfolders and picking only .yaml files etc.